### ISSUE TITLE
Update problem/Database issue

### ISSUE THREAD

**Issue by DarkObscurity** [johnnyq/itflow#204]
So let me rephrase my initial issue.

It seems that some things have/are going to change in the way the database is being created/filled over time due to changes being made during the development of IT-Flow.

This causes issues and in terms breaks function of the software when not handled properly.

I have noticed there is an update and a backup function in the web interface, however, the database backup is not really usable since the only current way to restore it, is to restore the file in the database host, this will cause issues if the database differs from what the web interface is working with.

The update function however doesn't seem to do much? I'm presented with a blank page which doesn't seem to do very much.. 
So currently the only viable way to get the latest updates for me is to download another copy of the Github and copy those files over the current web files and re-apply the permissions for the Apache user.

However, these methods both break the way IT-Flow works, so I have no idea if I'm maybe missing something? or if something isn't working correctly?

But I think it might be a good solution to have a restore backup function in the web interface in which you can upload a backup file from your previous installation. If this feature is capable of scanning the data from which version of the previous lay-outs then it could be able to inject the data in the correct order and lay-out back in the current database.

This way, if there is an update in which a database change has been made, people can always default to reinstalling IT-Flow (Which to be honest is really easy and done quickly) and then just import the old database from the backup function and then they are ready to go! 

I'm curious if this would be possible, or if I might be missing some crucial information here!

**Comment by DarkObscurity** [comment]
Reinstalling seems do have fixed the issue, but this makes me wonder if I have to reinstall for each update? 
Or are there certain steps to take?

**Comment by DarkObscurity** [comment]
As an update I downloaded the new version from Github and then copied those files over the existing files since the update button gives me a blank webpage. (After copying the files, I did restore the permissions to the apache system user)

**Comment by johnnyq** [comment]
Hi @DarkObscurity You are correct the update function currently does not work. I think it may help to put WIP in places where its not quite ready. I will start doing that to make it more clear. The easiest way to update the files is to use git. Git is extremely powful and know what has changed and will update accordingly.

To do an update go to the web directory where itflow is and type git pull

This will pull the latest changes down.

The one thing this will not update is if database changes are made, like field names, additions, deletions etc. I will usually put those in the commit logs stating there were DB Updates made.

The update function will do something similar it will use git to pull down the latest changes. I would also like it to scan the existing db and new db schema for changes and apply those changes.

I hope this helps.

**Comment by johnnyq** [comment]
1551ea7c9bf58fb9ac2574975f4b83a2190d7877

Basic Functionality has been added, just need to work on the DB Schema updates between commits

**Comment by johnnyq** [comment]
I will add the DB Schema Functionality change on the next DB Schema update. Now it should hopefully be a nice rolling release.

**Comment by DarkObscurity** [comment]
Sounds awesome! Much appreciated!

**Comment by johnnyq** [comment]
Your very welcome thanks for pointing that one out. I completely forgot about the upgrade, i'm sure there are more sections I started I haven't finished yet. Ill do a crawl through and mark those ones as work in progress
